sentry-core/src/scope/real.rs

```diff
@@ -357,4 +369,33 @@ impl Scope {
             session.update_from_event(event);
         }
     }
+
+    pub(crate) fn apply_propagation_context(&self, event: &mut Event<'_>) {
+        if event.contexts.contains_key("trace") {
+            return;
+        }
+
+        let context = TraceContext {
+            trace_id: self.propagation_context.trace_id,
+            span_id: self.propagation_context.span_id,
+            ..Default::default()
+        };
+        event.contexts.insert("trace".into(), context.into());
+    }
+
+    /// Returns the headers needed for distributed tracing.
+    pub fn iter_trace_propagation_headers(&self) -> TraceHeadersIter {
```

Comment:
I'd recommend returning an opaque type here; that way, you can change the underlying `TraceHeadersIter` struct to a different type at any time without it being a breaking API change.

```suggestion
    pub fn iter_trace_propagation_headers(&self) -> impl Iterator<Item = (&'static str, String)> {
```

Reply:
Nice, I've used `TraceHeadersIter` here just because that's what the other functions on `Scope`/`Transaction` use but `impl Iterator<Item = TraceHeader>` sounds better.

Reply:
Yeah, I mean long term (as in, before a 1.x release) I'd recommend updating those as well. For now, you can keep them as is to avoid a breaking change